src/Common/Entity/User.php

```diff
@@ -80,22 +92,52 @@ public function setBirthday(?\DateTime $birthday): void
     }
 
     /**
-     * @return string|null
+     * @return string
      */
-    public function getSex(): ?string
+    public function getGender(): string
     {
-        return $this->sex;
+        return $this->gender;
     }
 
-    /**
-     * @param string $sex
-     */
-    public function setSex(string $sex): void
+    public function setGender(string $gender): void
     {
-        if ($sex !== self::SEX_MALE && $sex !== self::SEX_FEMALE) {
-            throw new \InvalidArgumentException('Argument $sex is not valid');
+        if ($gender === self::SEX_MALE) {
+            trigger_error('the constant SEX_MALE is deprecated. use GENDER_MALE instead', E_USER_DEPRECATED);
+            $gender = self::GENDER_MALE;
+        }
+        
+        if ($gender === self::SEX_FEMALE) {
+            trigger_error('the constant SEX_FEMALE is deprecated. use GENDER_FEMALE instead', E_USER_DEPRECATED);
+            $gender = self::GENDER_FEMALE;
+        }
+            
+        $genders = [
```

Comment:
Instead of this you can also declare a constant of valid values: `const VALID_GENDERS = [self::GENDER_MALE, self::GENDER_FEMALE ....]`